### ISSUE TITLE
Anchor Bottom data missing from downloads

### ISSUE THREAD

**Issue by IanMiddlebrook** [BiologicalRecordsCentre/UKBMS-online#366]
Hi @DavidRoy @Gary-van-Breda 

While carrying out data validation, a Branch Manager has noticed that the data from one of her sites is not included in any of the relevant downloads - Summary Level, Section Level, Occurrences.

The site is Anchor Bottom (locationID=site:2441)

She can see on the Annual Summary table that the site has been recorded this year, and the Samples are included in the Samples download, but none of the butterfly records can be downloaded.

This seems to also be true for my own downloads from the Full Data Set, and also from the individual recorders who have submitted data for this transect.

Could you please establish why these data are missing from the downloads, and investigate if there are any other sites that this might be happening to?

Many thanks, Ian

**Comment by IanMiddlebrook** [comment]
Hi, I've just looked further at this, and there is more confusion.

The Summary Table for Anchor Bottom shows a full range of butterflies recorded across the year.
However, if you swap over to the Raw Data table, there appear to be no butterflies, and if I try to look at the Data Entry pages for this  site, there are no butterflies listed.
It seems as though all the butterfly data has been erased in some way, though still showing on the Annual Summary table?

**Comment by andrewvanbreda** [comment]
Hi @ianmiddlebrook (and @DavidRoy for reference)

This issue hasn't been assigned to me by David which is normally how I work, however I see it is marked as critical and it heavily overlaps with my current task of making sure the reporting data on BMS sites is clear and the reports are operating correctly.

Are you still having a problem with this?
I did try looking in the database, and I can't see any occurrences related to Anchor Bottom sections. Let me know if you need assistance with this

**Comment by IanMiddlebrook** [comment]
Hi @andrewvanbreda 

Yes, this site is still presenting an issue.
I can see that the samples still exist, both by looking at the Raw Data tab on the Annual Summary page, and by looking at 'My Walks' page in the recorders user accounts.
However, all associated butterfly records (occurrences) appear to have disappeared.

Thanks, Ian

**Comment by andrewvanbreda** [comment]
@IanMiddlebrook 
OK thanks for letting me know. I will attempt to work out what is happening.

**Comment by andrewvanbreda** [comment]
Hi @IanMiddlebrook ,

Thank you for the email discussion that we conducted outside of Github (to avoid filling up the Github thread too much)

Just a quick summary for anyone else reading this.

You managed to download the Anchor Bottom data for 2023, but after that ALL the data for Anchor Bottom vanished, including 2023 and 2024.

As the data has vanished, this made searching the audit logs for it very difficult as there are no IDs to search for.

However you have been able to send me some example data from 2023 so I can see what has happened to that, then we can assume the same has happened to the 2024 data and track it down.

I have located the 2023 data items you sent me.

So the plus side of this is the 2024 data is almost certainly in the database also (it is a little slow to search for (for reasons I won't go into here), but I can see some rows from 2024).

Anchor bottom actually has 11 sections, 7 of which have been deleted

These are the non-deleted ones
Anchor Bottom - S1
Anchor Bottom - S2
Anchor Bottom - S3
Anchor Bottom - S4

There are deleted
Anchor Bottom - S1
Anchor Bottom - S2
Anchor Bottom - S2
Anchor Bottom - S3
Anchor Bottom - S4
1453.1
1453.2

(I do not know what these last 2 are, they might be auto-generated names by the computer system).

So we can see that duplicates of sections 1 to 4 also exist in a deleted state.
Section 2 is also appearing twice in a deleted state.

So why wasn't I initially finding the occurrences when I first searched for them?

When a section is deleted, the system marks items such as samples and occurrences as deleted, but it is doing it in a strange way. The sub-samples are disconnected from the main sample, the link is removed (I do not know why it would do that)
This is why we can still see lots of main samples attached to anchor bottom, but not sub-samples or occurrences.

However enough information is retained where I think can write some SQl database code that can manually reconstruct the data enough for an export.
I guess in the longer term I should investigate changing UKBMS so it doesn't behave like this, but I will need to check with David if he wants me to do that, also check with Gary whether that is a good idea, maybe there is a valid reason why it behaves like that. It might even be a general database behaviour and not UKBMS fault at all.

In summary
- The data is there
- The data is reconstructable and could be exported
- I can't just get you the data immediately as there is that untangling to do.

Another thing that could be done is to correct the data in the database rather than just export it, but that is probably something I would have to check with people first.

I guess it would be useful to know what would the urgency be of me getting you the 2024 data?

**Comment by IanMiddlebrook** [comment]
Thanks @andrewvanbreda 
There is no urgency in getting the 2024 data now, as we've completed our analysis for 2024.

The urgency would be in ensuring that fresh data entered now for 2025 are saved securely and can be seen/reviewed by the recorder on the website.

**Comment by andrewvanbreda** [comment]
@IanMiddlebrook 
Ok, thanks for letting me know.
I will take a look at why this happened and take any required action and keep you in loop.

**Comment by andrewvanbreda** [comment]
Hi @IanMiddlebrook  (and @DavidRoy),

I have looked into this situation this morning.

My conclusion is the system behaved correctly.
It is a bit confusing for a project like UKBMS which has different levels in the data structures, deleting part of the structure such as a section can leave a situation where the data can appear confusing even if it is technically correct.
For example it is correct that the deleted section causes the "location_id" link on the section sub-sample to be cleared, but as the main transect sample isn't affected by this, it can make this confusing when querying the database.
Am checking with John van Breda about whether this clearance is strictly necessary or not, but I suspect it is, and is universal across projects...it is just more confusing when using sub-samples.

Although it behaved correctly, that doesn't mean we can't improve things, so for now I have altered the Delete Section warning to inform the user that unlike redrawing, this will immediately remove any records. Previously the system did not make it clear that deleting a section could have consequences.
To make this all a bit clearer for myself also, I will make a note about how to query the database correctly in this situation.

Could you let know if you are going to need that 2024 data at all?
If you do, I will ask if David if he wants to handle getting you that data internally and then I think we can start looking towards perhaps getting this issue closed.

Hope that helps

Andy

**Comment by IanMiddlebrook** [comment]
Thanks @andrewvanbreda 
Yes, we would like to recover the 2024 data if possible (and any other previous data that were deleted at the same time.
Though I guess the previous data could be re-entered, as we have those in our offline database.

Does that mean the recorders can enter data safely again this year?

**Comment by andrewvanbreda** [comment]
Hi @IanMiddlebrook,

As it can be hard to follow Github issues when they get too long,  so I will close this issue and raise a new Github issue for the data export and assign it to David to choose who should carry out the export, you will also be able to see progress there.

In terms of the safety of data entry on UKBMS.
There is nothing about this Github issue that puts into question the reliability of saving data on the system.
I was not present for the testing of the website originally, so for our peace of mind what I have done is record some temporary data and have attached screenshots of the data before saving, and also how it looks opened for editing.
We can see the data below is saved ok.

Note the website did go through a recent big upgrade, so you may find some differences on the site recently (such as the Explore Map has actually started again!).
The data entry form is the same though.


BEFORE (note the 0 in the total on the last row is correct, I did not click off the field before saving deliberately to test it worked in that situation)

<img width="827" alt="Image" src="https://github.com/user-attachments/assets/4b4d84ab-08e0-4121-b301-a4ecaf1d14ff" />

AFTER SAVE

<img width="763" alt="Image" src="https://github.com/user-attachments/assets/65a949b3-4405-44ad-afe5-5003270831bf" />

**Comment by andrewvanbreda** [comment]
The rest of this issue now now covered by
https://github.com/BiologicalRecordsCentre/UKBMS-online/issues/375
